test/e2e/usersignup_test.go

```diff
@@ -0,0 +1,90 @@
+package e2e
+
+import (
+	"context"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+)
+
+
+func TestUserSignup(t *testing.T) {
+	userSignupList := &v1alpha1.UserSignupList{}
+
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, userSignupList)
+	require.NoError(t, err, "failed to add custom resource scheme to framework")
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	err = ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err, "failed to initialize cluster resources")
+	t.Log("Initialized cluster resources")
+
+	namespace, err := ctx.GetNamespace()
+	require.NoError(t, err, "failed to get namespace where operator needs to run")
+
+	f := framework.Global
+	//client := f.Client.Client
+
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "host-operator", 1, operatorRetryInterval, operatorTimeout)
+	require.NoError(t, err, "failed while waiting for host operator deployment")
+
+	t.Log("host-operator is ready and in a running state")
+
+	// Create user signup
+	t.Logf("Creating UserSignup with namespace %s", namespace)
+	userSignup := newUserSignup(namespace, "foo")
+	err = f.Client.Create(context.TODO(), userSignup, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err)
+	t.Logf("user signup '%s' created", userSignup.Name)
+
+	// Confirm the UserSignup was created
+	err = verifyResources(t, f, namespace, userSignup)
+	require.NoError(t, err)
```

Comment:
Above we created UserSignup and it returns No error.  So in `verifyResources()` we may NOT need to verify UserSignup but we must verify creation of SecondaryResources in `verifyResources()`, I guess secondary resource is MUR here .. right ??

Reply:
@nurali-techie that's correct, my local test checks for MasterUserRecord. I haven't committed the changes yet because I'm still working through e2e test failures with Dipak's assistance.

Reply:
Tests are fixed now, and I've pushed my changes.